openmc/model/model.py

```diff
@@ -136,29 +169,19 @@ def run(self, **kwargs):
         Returns
         -------
         2-tuple of float
-            k_combined from the statepoint
+            Combined estimator of k-effective from the statepoint
 
         """
-
         self.export_to_xml()
 
         return_code = openmc.run(**kwargs)
 
         assert (return_code == 0), "OpenMC did not execute successfully"
 
-        statepoint_batches = self.settings.batches
+        n = self.settings.batches
         if self.settings.statepoint is not None:
             if 'batches' in self.settings.statepoint:
-                statepoint_batches = self.settings.statepoint['batches'][-1]
-        self.sp = \
-            openmc.StatePoint('statepoint.{}.h5'.format(statepoint_batches))
-
-        return self.sp.k_combined
-
-    def close(self):
-        """Close the statepoint and summary files
-        """
+                n = self.settings.statepoint['batches'][-1]
 
-        if self.sp is not None:
-            self.sp._f.close()
-            self.sp.summary._f.close()
+        with openmc.StatePoint('statepoint.{}.h5'.format(n)) as sp:
+            return sp.k_combined
```

Comment:
It works fine for me and I'm using 2.6.0, but then again I haven't run into as many issues with statepoints as you have.